src/wp-includes/class-wp-hook.php

```diff
@@ -302,7 +302,7 @@ public function apply_filters( $value, $args ) {
 				}
 
 				// Avoid the array_slice() if possible.
-				if ( 0 == $the_['accepted_args'] ) {
+				if ( 0 === $the_['accepted_args'] ) {
```

Comment:
- While this continues to avoid `array_slice()` for `(int) 0`, the loose comparison meant that `(string) '0'` (and anything that evaluated to `0`, also avoided `array_slice()`.
- With the strict comparison, `(string) '0'` now passes through to `else`, and `array_slice()` is called.

`$accepted_args` is documented as `int`, and I think it would be incredibly rare for the following code:
```php
add_filter( 'hook_name', 'callback_name', $priority, '0' );
```

However, it may be worth considering whether the following check would prevent a performance regression for those (very rare) cases:
```php
if ( 0 === $the_['accepted_args'] || '0' === $the_['accepted_args'] ) {
```

Or, for full BC:
```php
if ( 0 === (int) $the_['accepted_args'] ) {
```

@mukeshpanchal27 Thoughts?

Reply:
Thanks @costdev.

The second approach, in my opinion, is a better way to solve the problem.

```
if ( 0 === (int) $the_['accepted_args'] ) {
```